tests/rptest/services/redpanda.py

```diff
@@ -960,6 +960,56 @@ def signal_redpanda(self, node, signal=signal.SIGKILL, idempotent=False):
         """
         pass
 
+    def set_cluster_config(self,
+                           values: dict,
+                           expect_restart: bool = False,
+                           admin_client: Optional[Admin] = None,
+                           timeout: int = 10):
+        """
+        Update cluster configuration and wait for all nodes to report that they
+        have seen the new config.
+
+        :param values: dict of property name to value. if value is None, key will be removed from cluster config
+        :param expect_restart: set to true if you wish to permit a node restart for needs_restart=yes properties.
+                               If you set such a property without this flag, an assertion error will be raised.
+        """
+        pass
+
+    def node_id(self, node, force_refresh=False, timeout_sec=30):
+        """
+        Returns the node ID of a given node. Uses a cached value unless
+        'force_refresh' is set to True.
+
+        NOTE: this is not thread-safe.
+        """
+        pass
+
+    def rolling_restart_nodes(self,
```

Comment:
This and some other methods (e.g. `metric_sum()`) seem like a higher level algorithms to me that do not need overrides in containerized/k8ized descendants, but rather an implementation in the base class using other virtual methods like `start_node()` or `metrics_sample()`.

IMO the most feasible approach to the DT/k8s adoption would be to select the simplest of all the tests in scope, and begin with porting that test only, rather than trying to cover them all at once. That would likely give the most basic abstraction required. Expanding from there to more and more tests will make it clear which methods require overloading and which do not.

Reply:
looks like this method makes use of [RollingRestarter](https://github.com/redpanda-data/redpanda/blob/dev/tests/rptest/services/rolling_restarter.py#L15) class which is tightly-coupled with the `RedpandaService`. i'll work out what the deps are and bring them into this base class.

i've created a follow-on issue #9709 to track implementation of the k8s-specific methods which i'll get to after this PR merges and the k8s test infrastructure is up (tracked in https://github.com/redpanda-data/devprod/issues/650)

Reply:
i've moved this impl into the base class with d17853eb247a65533e530548c9d511a0ffd7a576

Reply:
update: i've force-pushed and this is now 409004d4b28e150a083c847f5bbba7bd0aa4e9f0